### PR TITLE
README and Makefile cleanups

### PR DESCRIPTION
**Pull request by myleshorton**
This breaks out the release details from the README @xiam and @oxtoacart since that only applies to Lantern developers actually creating releases and goes into detail that's unusual for a Makefile. I also added simple instructions for building "the old fashioned way" with just `go build`. Finally, I added back the `make docker` target in the Makefile.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -164,13 +164,28 @@ define docker-up
...
+	echo "Building docker..." && \
+	DOCKER_CONTEXT=.$(DOCKER_IMAGE_TAG)-context && \
+	mkdir -p $$DOCKER_CONTEXT && \
+	cp Dockerfile $$DOCKER_CONTEXT && \
+	echo "Calling docker build..." && \
+	docker build -t $(DOCKER_IMAGE_TAG) $$DOCKER_CONTEXT && \
+	echo docker run `echo $(DOCKER_VOLS) | xargs` -t $(DOCKER_IMAGE_TAG) /bin/bash -c 'cd /lantern && VERSION="'$$VERSION'" HEADLESS="'$$HEADLESS'" BNS_CERT_PASS="'$$BNS_CERT_PASS'" make $*' && \
+	docker run `echo $(DOCKER_VOLS) | xargs` -t $(DOCKER_IMAGE_TAG) /bin/bash -c 'cd /lantern && VERSION="'$$VERSION'" HEADLESS="'$$HEADLESS'" BNS_CERT_PASS="'$$BNS_CERT_PASS'" make $*';
+
+docker-test: system-checks
```
Comment (oxtoacart) [inline Makefile]: @myleshorton Why does this target need to duplicate the code on lines 173 to 180?

File: Makefile [hunk 2]
```diff
@@ -164,13 +164,28 @@ define docker-up
 	fi
 endef
 
+docker: docker-
```
Comment (oxtoacart) [inline Makefile]: This target fails when I run it, with:

```
cc  -s     -o .o
cc: fatal error: no input files
compilation terminated.
<builtin>: recipe for target '.o' failed
make: *** [.o] Error 4
make: *** [docker-.o] Error 2
```

What's the purpose of this target?

File: README.md [hunk 3]
```diff
@@ -13,268 +13,39 @@
...
-### Building the docker image
-
-In order to build the docker image open a terminal, `cd` into the
-`lantern` project and execute `make docker`:
+To build and run Lantern desktop, just do:
 
 ```sh
+git clone https://github.com/getlantern/lantern.git
 cd lantern
 make docker
```
Comment (oxtoacart) [inline README.md]: I think you mean `make lantern` or `make docker-lantern`?

File: README.md [hunk 4]
```diff
@@ -13,268 +13,39 @@
...
-
-After you're satisfied with a beta version, it will be time to promote beta
-packages to production and to publish the packages for auto-updates:
-
-```
-VERSION=2.0.0-beta5 GH_TOKEN=$GITHUB_TOKEN make release
+git clone https://github.com/getlantern/lantern.git
+cd lantern
+source setenv.bash
+go build -o lantern github.com/getlantern/flashlight/main
```
Comment (oxtoacart) [inline README.md]: This won't work if you're missing the assets.  `make lantern` is safer.

Comment (myleshorton) [inline README.md]: Ah true. A little unfortunate that's required even when just running locally, but that's a different issue! I'll remove that version.

File: Makefile (hunk 1 above)
Comment (myleshorton) [inline Makefile]: Oh sorry -- committed docker-test by mistake. Will fix!

File: Makefile (hunk 2 above)
Comment (myleshorton) [inline Makefile]: I was actually just trying to make everything compatible with the prior README, which included the `make docker` step that no longer worked. With just `make lantern` we can eliminate that step, however.

**Comment by myleshorton** [comment]
Hey how's this look now @oxtoacart?

**Comment by oxtoacart** [comment]
LGTM.  I made one small addition about using clean-desktop - https://github.com/getlantern/lantern/commit/e6b20a5e84696ec7bc8b1ef028771811c863f444

**Pull request merged**